conda_recipe_manager/parser/recipe_reader.py

```
@@ -510,50 +403,22 @@ def _render_jinja_vars(self, s: str) -> JsonType:
-                # that will normalize it back to a string. But for this to work, we must perform the `split` before a
-                # `join` or an index access.
-                if split_match:
-                    value = value.split(split_match.group(2))  # type: ignore[assignment]
-                if join_match:
-                    # The index of the string that we join on changes depending on which form of the function is used.
-                    # NOTE: `.groups()` does not return the top-level 0th grouping, so the indices are 1-off.
-                    join_str_idx = 3 if join_match.groups()[2] is not None else 2
-                    value = join_match.group(join_str_idx).join(value)
-                if idx_match:
-                    idx = int(cast(str, idx_match.group(2)))
-                    # From our research, it looks like string indexing on JINJA variables is almost exclusively used
-                    # get the first character in a string. If the index is out of bounds, we will default to the
-                    # variable's value as a fall-back. Although rare, negative indexing is supported.
-                    if -len(value) <= idx < len(value):
-                        value = value[idx]
-                if replace_match:
-                    value = value.replace(replace_match.group(2), replace_match.group(3))
-                s = s.replace(match, value)
+            expression = match[start_idx:-2].strip()
+            context = {k: self.get_variable(k) for k in self._vars_tbl}
+            # If the expression can't be evaluated, skip it.
+            try:
+                env = Environment(undefined=StrictUndefined)  # type: ignore[misc]
+                compiled_expression = env.compile_expression(expression)
+                result = compiled_expression(**context)  # type: ignore[misc]
+            except Exception:  # pylint: disable=broad-exception-caught
+                continue
```

Should we be logging these silent failures?